full-service/src/db/mod.rs

```diff
@@ -43,15 +47,28 @@ pub fn b58_decode(b58_public_address: &str) -> Result<PublicAddress, WalletDbErr
     Ok(PublicAddress::try_from(pubaddr_proto).unwrap())
 }
 
+/// A struct encapsulating a single connection, which can also be used for encryption and logging.
```

Comment:
@eranrund - please take a look at the WalletDbConnManager and let me know what you think!